File: src/main/java/freemusic/music/service/DownloadService.java
package freemusic.music.service;

import freemusic.music.service.download.KgDownload;
import freemusic.music.service.download.MusicDownload;
import freemusic.music.service.download.QQDownload;
import freemusic.music.service.download.WyDownload;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class DownloadService {

    public String getDownloadUrl(String type, String quality, String id) {
        MusicDownload musicDownload = getService(type);
        String url = musicDownload.getDownloadUrl(id, quality);
        return url;
    }

    public String getMvUrl(String type, String quality, String id) {
        MusicDownload musicDownload = getService(type);
        String url = musicDownload.getMvUrl(id, quality);
        return url;
    }


    private MusicDownload getService(String type) {
        MusicDownload musicDownload = null;
        if (type.equals("1")) {
            musicDownload = qqDownload;
        } else if (type.equals("2")) {
            musicDownload = wyDownload;
        } else if (type.equals("3")) {
            musicDownload = kgDownload;
        }
        return musicDownload;
    }

    @Autowired
    private QQDownload qqDownload;

    @Autowired
    private KgDownload kgDownload;

    @Autowired
    private WyDownload wyDownload;


}

File: src/main/java/freemusic/music/service/DownloadService.kt
package freemusic.music.service

import freemusic.music.service.download.KgDownload
import freemusic.music.service.download.MusicDownload
import freemusic.music.service.download.QQDownload
import freemusic.music.service.download.WyDownload
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class DownloadService {

    fun getDownloadUrl(type: String, quality: String, id: String): String {
        val musicDownload = getService(type)
        return musicDownload.getDownloadUrl(id, quality)
    }

    fun getMvUrl(type: String, quality: String, id: String): String {
        val musicDownload = getService(type)
        return musicDownload.getMvUrl(id, quality)
    }


    private fun getService(type: String): MusicDownload {
        return when (type) {
            "1" -> qqDownload
            "2" -> wyDownload
            "3" -> kgDownload
            else -> qqDownload
        }
    }

    @Autowired
    private val qqDownload: QQDownload = QQDownload()

    @Autowired
    private val kgDownload: KgDownload = KgDownload()

    @Autowired
    private val wyDownload: WyDownload = WyDownload()
}